app/models/bookable_slot.rb

```diff
@@ -14,6 +14,16 @@ def pm?
     PM.pm?(self.end)
   end
 
+  def appointments
+    scope = Appointment.where(location_id: schedule.location_id)
+
+    if am?
+      scope.where('proceeded_at >= ? AND proceeded_at < ?', start_at, end_at)
```

Comment:
Since the end of the AM period and the start of the PM period is the same, doesn't a similar clause apply below too, or am I missing the obvious?